File: Scoreboard/Scoreboard.java
public class Scoreboard {

    private int activeTeam;
    private String teamName1;
    private String teamName2;
    private int score1;
    private int score2;

    public Scoreboard(String team1, String team2) {
        activeTeam = 1;
        teamName1 = team1;
        teamName2 = team2;
    }
    
    public String getScore() {
        if (activeTeam == 1) {
            return (score1 + "-" + score2 + "-" + teamName1);
        } else if (activeTeam == 2) {
            return (score1 + "-" + score2 + "-" + teamName2);
        }
        return null;
    }

    public void recordPlay(int point) {
        if (point == 0) {
            if (activeTeam == 1) {
                activeTeam = 2;
            } else if (activeTeam == 2) {
                activeTeam = 1;
            }
        } else if (activeTeam == 1) {
            score1 += point;
        } else if (activeTeam == 2) {
            score2 += point;
        }
    }
}

File: Scoreboard/Scoreboard.py
class Scoreboard:
    # construct class so that args contains all inputs after the team names
    def __init__ (self, team1, team2, *args):
        self.team1 = team1
        self.team2 = team2
        self.args = args

    def winner(self):
        score1 = 0
        score2 = 0
        teamScoring = 1
        # for each input check if team in play is swapped and add score
        for arg in self.args[0]:
            if teamScoring == 1:
                if int(arg) == 0:
                    teamScoring = 2
                else:
                    score1 += int(arg)
            else:
                if int(arg) == 0:
                    teamScoring = 1
                else:
                    score2 += int(arg)
        # return team with most points or whether game is a tie
        if score1 > score2:
            return self.team1
        elif score2 > score1:
            return self.team2
        elif score1 == score2:
            return "tie"